### PR TITLE
Migrated TFS 2010 Bug Reporting Guide

### PR DESCRIPTION
**Pull request by gocGrabinski**
Added bug reporting guide for TFS 2010

### CODE REVIEW

File: source-control/Bug_Reporting_TFS_2010.md [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+  * [1.6 Bug Reporting Diagram](#Bug_Reporting_Diagram)
+  * [1.7 Team Queries](#Team_Queries)
+    * [1.7.1 Active Bugs](#Active_Bugs)
+    * [1.7.2 Bugs to be tested and/or closed](#Bugs_to_be_tested_and.2For_closed)
+  * [1.8 Build Definition Naming Convention (.Net Stack)](#Build_Definition_Naming_Convention_.28.Net_Stack.29)
+  * [1.9 Testing Complex or Multi-Application Solutions](#Testing_Complex_or_Multi-Application_Solutions)
+    * [1.9.1 Option 1: Provide access to your TFS](#Option_1:_Provide_access_to_your_TFS)
+    * [1.9.2 Option 2: Merge into 1 Team Project](#Option_2:_Merge_into_1_Team_Project)
+    * [1.9.3 Option 3: Create a Central Team Project](#Option_3:_Create_a_Central_Team_Project)
+  * [1.10 Template for "Details = Steps to Reproduce" Field](#Template_for_.22Details_.3D_Steps_to_Reproduce.22_Field)
```
Comment (s-laugh) [inline source-control/Bug_Reporting_TFS_2010.md]: Not all these links work. I don't think we need all of them either; probably just the first level.

File: source-control/Bug_Reporting_TFS_2010.md [hunk 2]
```diff
@@ -0,0 +1,520 @@
...
+The Validator is usually the same person as the Reporter of the issue but can be another person.  
+The Validator will set the bug to the Closed state.
+
+### Fields of a Bug
+
+The fields available in the Bug work item type have been designed to help with tracking, reporting and to collaborate efficiently. To document the defect, the fields need to be properly populated and can be updated during the bug reporting process as details of the defect unfold. Incorrect values can impact the results of queries, which are used to list and assign work to individuals.
+
+Below are the recommendations for the main fields of the Bug Work Item. Fields marked with an  are the most crucial to populate.
+
+[![image](/gcwiki/images/f/f7/BugWindow.JPG)](/wiki/File:BugWindow.JPG "image")
```
Comment (s-laugh) [inline source-control/Bug_Reporting_TFS_2010.md]: The images aren't loading. You'll want to upload them to the `assets` folder.

File: source-control/Bug_Reporting_TFS_2010.md [hunk 3]
```diff
@@ -0,0 +1,520 @@
...
+
+* Identifies who in the project team is responsible for taking action on the work item.
+* Generic lists should be used when assigning to the Dev Coordinator and TST Coordinator.
+* The Dev and TST Coordinators will select the specific resource once it's decided who will action the work item.
+
+#### States and Reasons
+
+Below you will find the States and their available Reasons for the work item type Bug in the default Agile Template.
+
+[![image](/gcwiki/images/f/f6/BugReasonsStates2.JPG)](/wiki/File:BugReasonsStates2.JPG "image")
```
Comment (s-laugh) [inline source-control/Bug_Reporting_TFS_2010.md]: Image not loading. Should upload it to `assets` folder.

File: source-control/Bug_Reporting_TFS_2010.md [hunk 4]
```diff
@@ -0,0 +1,520 @@
...
+
+#### Area
+
+* Identifies the application, component or module impacted
+* It can be modified when the resolution of the bug is found. Example: the area selected at the creation of the Bug was "MSCBA Web" but after investigation the issue was in the Stored Procedure and the area was modified to reflect this.
+* Do not use Area to group and/or track bugs raised by Security/Performance testing. The field "Classification" should be used for this.
+* Guidelines can be found in the \[[TFS Project Setup Recommendation](Bug_Reporting_TFS_2010.md\]
```
Comment (s-laugh) [inline source-control/Bug_Reporting_TFS_2010.md]: This pointed to a dead link, probably just remove it.
```suggestion
```

File: source-control/Bug_Reporting_TFS_2010.md [hunk 5]
```diff
@@ -0,0 +1,520 @@
...
+
+#### Iteration:
+
+* Iterations are a way to identify the cycle/release/milestone/sprint you are working on.
+* Guidelines can be found in the \[[TFS Project Setup Recommendation](Bug_Reporting_TFS_2010.md\]
```
Comment (s-laugh) [inline source-control/Bug_Reporting_TFS_2010.md]: This pointed to a dead link, probably just remove it.
```suggestion
```

File: source-control/Bug_Reporting_TFS_2010.md [hunk 6]
```diff
@@ -0,0 +1,520 @@
...
+3. Medium: (Default) A defect that causes the system to produce incorrect, incomplete or inconsistent results.
+4. Low: A minor or cosmetic defect that has acceptable workarounds to achieve required results.
+
+#### System Info:
+
+* Tells us in which Build a bug was found and in which Build the correction was made.
+* These fields are used by the Queries to determine which bugs are ready for testing. Useful when a new deploy of the application is made.
+* The values for the "Found In Build" and "Integrated In Build" fields should only be selected from their respective drop downs.
+  * These fields are automatically populated when the Build Server is used.
+  * Optionally they can be populated manually via the Global List. \[[Instructions to populate manually](Bug_Reporting_TFS_2010.md\].
```
Comment (s-laugh) [inline source-control/Bug_Reporting_TFS_2010.md]: Another previously dead link.
```suggestion
  * Optionally they can be populated manually via the Global List.
```

File: source-control/Bug_Reporting_TFS_2010.md [hunk 7]
```diff
@@ -0,0 +1,520 @@
...
+  * More details are provided to explain why it was rejected.
+  * Fields can be updated if necessary
+  * The cycle starts over.
+* In all situations, all the Fields should be revised and updated if needed.
+
+### Bug Reporting Diagram
+
+[![Bugprocesstop.JPG](/gcwiki/images/thumb/1/11/Bugprocesstop.JPG/300px-Bugprocesstop.JPG)](/wiki/File:Bugprocesstop.JPG)
+
+[![Bugprocessbottom.JPG](/gcwiki/images/thumb/8/80/Bugprocessbottom.JPG/300px-Bugprocessbottom.JPG)](/wiki/File:Bugprocessbottom.JPG)
```
Comment (s-laugh) [inline source-control/Bug_Reporting_TFS_2010.md]: Images not loading. Should upload it to `assets` folder.

File: source-control/Bug_Reporting_TFS_2010.md [hunk 8]
```diff
@@ -0,0 +1,520 @@
...
+
+### Team Queries
+
+The following "Team Queries" are executed to produce a list of the "Active Bugs" or Bugs that are "Ready for testing" for a specific "Iteration".
+
+For future releases (Iterations), the queries can be copied and then the "Iteration" filter updated. They can also be copied by team members to their "My Queries" section and add a clause to specify work items assigned to them.
+
+More filters could be added to better meet your needs example: The "Area" could be added if you have multiple components released in this same iteration and need to isolate the work items.
+
+Visit our "[TFS Project Setup Recommendation](Bug_Reporting_TFS_2010.md" for more details on Team Queries and naming convention.
```
Comment (s-laugh) [inline source-control/Bug_Reporting_TFS_2010.md]: Point to updated/migrated doc
```suggestion
Visit our [TFS 2015 Use and Configuration Recommendation](https://esdc-devcop.github.io/recommendations/tools/tfs2015.html) for more details on Team Queries and naming convention.
```

File: source-control/Bug_Reporting_TFS_2010.md [hunk 9]
```diff
@@ -0,0 +1,520 @@
...
+
+#### Active Bugs
+
+This query will be executed mostly by the Dev Coordinator to list new bugs and bugs that have been re-activated for a particular iteration.
+
+[![image](/gcwiki/images/7/71/ActiveBugs_Query.JPG)](/wiki/File:ActiveBugs_Query.JPG "image")
```
Comment (s-laugh) [inline source-control/Bug_Reporting_TFS_2010.md]: Image not loading. Should upload it to assets folder.

File: source-control/Bug_Reporting_TFS_2010.md [hunk 10]
```diff
@@ -0,0 +1,520 @@
...
+
+#### Bugs to be tested and/or closed
+
+This query will be executed mostly by the TST Coordinator to list bugs that have been resolved and are ready to be tested or be closed. At each deploy to the SysTest vertical, the clause "and Integration Build <=" will have to be modified to include the latest build deployed.
+
+[![image](/gcwiki/images/7/73/ReadytotestBugs_Query.JPG)](/wiki/File:ReadytotestBugs_Query.JPG "image")
```
Comment (s-laugh) [inline source-control/Bug_Reporting_TFS_2010.md]: Image not loading. Should upload it to assets folder.

**Review by s-laugh — CHANGES_REQUESTED** [review]
Looks good, mostly just broken links and images.

**Comment by gocGrabinski** [comment]
All done just waiting for final review to complete PR.

File: source-control/Bug_Reporting_TFS_2010.md [hunk 11]
```diff
@@ -0,0 +1,520 @@
...
+   **PROBLEM**
+   A short summary of the bug (i.e., 1 or 2 sentences)
+
+   **URL**: URL where the issue was found
+
+   **Environment**: Environment where the issue was found (TST, UAT...)
+   **Date of Occurrence:** If different then the date the bug was created
+   **USER:** Include user that was used when this bug was found and, if pertinent; include information on user role and privileges.  Remove this section
+   if not applicable.
+   **Test Script:** If applicable list the automated test used to identify or reproduce the bug
```
Comment (s-laugh) [inline source-control/Bug_Reporting_TFS_2010.md:472]: Not Blocking: These are being bundled into a single paragraph, I'm not sure if that was intend or not.

**Review by s-laugh — CHANGES_REQUESTED** [review]
Images look good now!
Still some broken links from the previous review, and one additional thing I noticed

File: source-control/Bug_Reporting_TFS_2010.md (hunk 11 above)
Comment (gocGrabinski) [reply source-control/Bug_Reporting_TFS_2010.md:472]: As intended.

**Comment by gocGrabinski** [comment]
Should be good to go now.

File: source-control/Bug_Reporting_TFS_2010.md (hunk 6 above)
Comment (s-laugh) [reply source-control/Bug_Reporting_TFS_2010.md]: This dead link is still present.

File: source-control/Bug_Reporting_TFS_2010.md (hunk 8 above)
Comment (s-laugh) [reply source-control/Bug_Reporting_TFS_2010.md]: This link still isn't updated.

**Comment by s-laugh** [comment]
Just 2 dead links got missed.

**Comment by gocGrabinski** [comment]
Done ( I hope ) :-)

**Comment by s-laugh** [comment]
Sorry, I made that a little confusing, this one actually has an updated link that should be used https://esdc-devcop.github.io/recommendations/tools/tfs2015.html

https://github.com/esdc-devcop/guides/pull/13#discussion_r349702608

**Comment by gocGrabinski** [comment]
Done ( I hope ) :-) part II

**Review by s-laugh — DISMISSED** [review]

**Comment by s-laugh** [comment]
Looks like your first two commits didn't get signed, which is blocking the merge... :(
Hopefully you figured out your local Git issues, because you can run this to "re-sign" them https://superuser.com/a/1123928

**Comment by s-laugh** [comment]
Scratch that... you did sign them, you just haven't uploaded you public key to GitHub.
https://help.github.com/en/enterprise/2.15/user/articles/adding-a-new-ssh-key-to-your-github-account

**Comment by gocGrabinski** [comment]
Added my public key - but you sure my commit was signed before ? I used git hub web editor and commit. Not sure those would sign the commit and if it did what key did it sign it with ?

**Comment by gocGrabinski** [comment]
@s-laugh - got my local git back into proper (?) working order. Only change is that I moved the images into sub directory. Should be able to merge as I signed the commit locally ... let's see how this goes.

**Review by s-laugh — APPROVED** [review]

**Pull request merged**